tests/testthat/test-plot_top_k.R

```diff
@@ -39,20 +39,20 @@ test_that("plot_top_k and predict_top_k fail when they should", {
                          nmc = 20, save_aug = TRUE, seed = 1L)
 
   pred <- predict_top_k(bmm, burnin = 4, k = 3)
+
   expect_equal(
-    pred[13:16, ],
-    structure(list(assessor = c(3, 3, 3, 4), item = c("Item 3", "Item 4",
-                                                      "Item 5", "Item 1"), prob = c(1, 0.3125, 0, 0.5625)), row.names = c("3.3",
-                                                                                                                          "3.4", "3.5", "4.1"), class = "data.frame")
+    pred[order(pred$assessor, pred$item), ][13:16, ],
+    structure(list(assessor = c(3, 3, 3, 4), item = c("Item 3", "Item 4", "Item 5", "Item 1"),
+                   prob = c(1, 0.3125, 0, 0.5625)),
+              row.names = c(119L, 177L, 235L, 4L), class = "data.frame")
```

Comment:
Isn't it a problem that the `row.names` change?

Reply:
Good point! I'm a bit divided. `row.names` are not used consistently throughout the package, and I don't think it's common to use `row.names`. On the other hand, the previous rownames were quite informative (`assessor.item`), so I'll see if I'm able to get them back to what they were.

Reply:
Turns out the row names in the current release are misleading, becauase they don't correspond to `assessor.item`. Hence it seems best to just remove them.

``` r
library(BayesMallows)
sessionInfo(package = "BayesMallows")
#> R version 4.1.2 (2021-11-01)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19042)
#> 
#> Matrix products: default
#> 
#> locale:
#> [1] LC_COLLATE=English_United States.1252 
#> [2] LC_CTYPE=English_United States.1252   
#> [3] LC_MONETARY=English_United States.1252
#> [4] LC_NUMERIC=C                          
#> [5] LC_TIME=English_United States.1252    
#> 
#> attached base packages:
#> character(0)
#> 
#> other attached packages:
#> [1] BayesMallows_1.1.0
#> 
#> loaded via a namespace (and not attached):
#>  [1] Rcpp_1.0.7        knitr_1.36        magrittr_2.0.1    grDevices_4.1.2  
#>  [5] R.cache_0.15.0    rlang_0.4.12      fastmap_1.1.0     fansi_0.5.0      
#>  [9] stringr_1.4.0     styler_1.7.0      highr_0.9         rbibutils_2.2.4  
#> [13] tools_4.1.2       utils_4.1.2       xfun_0.28         R.oo_1.24.0      
#> [17] utf8_1.2.2        withr_2.4.3       htmltools_0.5.2   stats_4.1.2      
#> [21] ellipsis_0.3.2    datasets_4.1.2    yaml_2.2.1        digest_0.6.29    
#> [25] tibble_3.1.6      base_4.1.2        lifecycle_1.0.1   crayon_1.4.2     
#> [29] purrr_0.3.4       Rdpack_2.1.2      graphics_4.1.2    vctrs_0.3.8      
#> [33] R.utils_2.11.0    fs_1.5.1          glue_1.6.2        evaluate_0.14    
#> [37] rmarkdown_2.11    reprex_2.0.1      stringi_1.7.6     compiler_4.1.2   
#> [41] pillar_1.6.4      methods_4.1.2     R.methodsS3_1.8.1 pkgconfig_2.0.3
library(stringr)
model_fit <- compute_mallows(preferences = beach_preferences,
                             save_aug = TRUE)
#> Generating transitive closure of preferences.
#> Generating initial ranking.
model_fit$burnin <- 1000
probs <- predict_top_k(model_fit)

head(probs)
#>     assessor    item  prob
#> 1.1        1  Item 1 0.046
#> 1.2        1 Item 10 0.245
#> 1.3        1 Item 11 0.967
#> 1.4        1 Item 12 0.018
#> 1.5        1 Item 13 0.000
#> 1.6        1 Item 14 0.000

all.equal(
  str_extract(row.names(probs), "[0-9]+$"),
  str_extract(probs$item, "[0-9]+$")
)
#> [1] "840 string mismatches"
```

<sup>Created on 2022-05-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>